src/cl/ucc_cl.h

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2020.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "config.h"
+
+#ifndef UCC_CL_H_
+#define UCC_CL_H_
+
+#include "api/ucc.h"
+#include "core/ucc_lib.h"
+#include "utils/ucc_component.h"
+#include "utils/ucc_parser.h"
+
+typedef struct ucc_cl_lib   ucc_cl_lib_t;
+typedef struct ucc_cl_iface ucc_cl_iface_t;
+
+typedef struct ucc_cl_lib_config {
+    /* Log level above which log messages will be printed */
+    ucc_log_component_config_t log_component;
+    /* Team library priority */
+    int                        priority;
+} ucc_cl_lib_config_t;
+
+extern ucc_config_field_t ucc_cl_lib_config_table[];
+
+typedef struct ucc_cl_iface {
+    ucc_component_iface_t          super;
+    int                            priority;
+    ucc_lib_attr_t                 attr;
+    ucc_config_global_list_entry_t cl_lib_config;
```

Comment:
shouldn't we move this into "super"? presumable any component will have a global list entry for config...

Reply:
super here is  a generic superclass for any "component" interfaces - cl, tl, maybe others. This struct here IS a top superclass for all the later CL ifaces, so it stores cl_lib_config - yes, any  component will inherit from cl_iface and have global list entry.

Reply:
What i meant was we should move `ucc_config_global_list_entry_t` into `ucc_component_iface_t` (and rename the member). This way not just CL components have config, but rather every component (incl. TL).

Reply:
But does any component should have an env var configuration ?

Reply:
In addition, the CL component iface will also have another config_list_entry: for context config. Remember we have to config_read APIs in UCC: for "lib" and for "context". Putting two configs into generic component iface looks weird.